### PR TITLE
KIALI-817 Change node shape based on namespace.

### PR DESCRIPTION
**Pull request by cfcosta**
This is the ui counterpart to [kiali/kiali#252](https://github.com/kiali/kiali/pull/252), so check there if you need information.

### CODE REVIEW

**Comment by jotak** [comment]
@cfcosta , it seems that changing colors for namespaces would not play well with another ongoing topic that is displaying health on nodes. The current plan was to change node border colors according to health. Looks like we're conflicting here? Too many colored information would, I think, make everything confusing.

See also https://issues.jboss.org/browse/KIALI-785 
@theute , @serenamarie125 what's your opinion?

**Comment by theute** [comment]
Agreed, do we really need a new color here ? 
(IMO the color line of the shape might reflect the health of the whole namespace, but we're not there yet)

**Comment by cfcosta** [comment]
@jshaughn can you review this, please?

File: src/actions/ServiceGraphDataActions.ts [hunk 1]
```diff
@@ -42,7 +42,8 @@ const decorateGraphData = (graphData: any) => {
       isGroup: undefined,
       isRoot: undefined,
       isUnused: undefined,
-      hasMissingSidecars: undefined
+      hasMissingSidecars: undefined,
+      isOutsideNamespace: undefined
```
Comment (jshaughn) [inline src/actions/ServiceGraphDataActions.ts]: You need to rebase to fix conflicts (e.g. 'hasMissingSidecar's is now 'hasMissingSC').  With respect to the server PR, 'isOutsideNamespace' is now just 'isOutside'

File: src/components/CytoscapeGraph/graphs/GraphStyles.ts [hunk 2]
```diff
@@ -31,15 +31,23 @@ export class GraphStyles {
         css: {
           // color: PfColors.Black,
           content: (ele: any) => {
+            const name = ele.data('service') || ele.data('id');
+            const namespace = name.split('.')[1];
+            const service = name.split('.')[0];
```
Comment (jshaughn) [inline src/components/CytoscapeGraph/graphs/GraphStyles.ts]: you should now be able use ele.data('serviceName') and ele.data('namespace') and avoid the split completely.

File: src/components/CytoscapeGraph/graphs/GraphStyles.ts [hunk 3]
```diff
@@ -31,15 +31,23 @@ export class GraphStyles {
...
             const version = ele.data('version');
+
+            if (ele.data('isOutsideNamespace')) {
```
Comment (jshaughn) [inline src/components/CytoscapeGraph/graphs/GraphStyles.ts]: again, 'isOutside'

File: src/components/CytoscapeGraph/graphs/GraphStyles.ts [hunk 4]
```diff
@@ -70,6 +78,12 @@ export class GraphStyles {
           shape: 'diamond'
         }
       },
+      {
+        selector: 'node[isOutsideNamespace]',
```
Comment (jshaughn) [inline src/components/CytoscapeGraph/graphs/GraphStyles.ts]: one more, 'isOutside'

**Comment by jshaughn** [comment]
Adding DNM label.  We do this whenever the ui PR first requires merge of a server PR.

File: src/components/CytoscapeGraph/graphs/GraphStyles.ts [hunk 5]
```diff
@@ -40,15 +40,22 @@ export class GraphStyles {
         css: {
           // color: PfColors.Black,
           content: (ele: any) => {
+            const service = ele.data('serviceName') || ele.data('service').split('.')[0];
+            const namespace = ele.data('namespace');
             const version = ele.data('version');
+
+            if (ele.data('isOutside')) {
```
Comment (josejulio) [inline src/components/CytoscapeGraph/graphs/GraphStyles.ts]: I think this should be *after* the line 53
```
if (!ele.data('showNodeLabels')) {
   return '';
}
```

else the label will be showing even though showNodeLabels is false

Comment (cfcosta) [reply src/components/CytoscapeGraph/graphs/GraphStyles.ts]: True! Fixed.

**Comment by serenamarie125** [comment]
I don't see a screenshot of what the PR explains ... do you have a screenshot of what it looks like?  What shape are you using?  If there are objects from multiple namespaces, are you using the same shape? or different ones?

**Pull request merged**